Highly accurate protein structure prediction with AlphaFold
J Jumper, R Evans, A Pritzel, T Green, M Figurnov, O Ronneberger, et al
Nature volume 596, pages 583–589 (2021)Cite this article

# Abstract

Proteins are essential to life, and understanding their structure can
facilitate a mechanistic understanding of their function. Through an enormous
experimental effort1,2,3,4
* the structures of around 100,000 unique proteins have been determined5, but
  this represents a small fraction of the billions of known protein seqs,7
* Predicting the three-dimensional structure that a protein will adopt based
  solely on its amino acid sequence—the structure prediction component of the
  ‘protein folding problem’8—has been an important open research problem for
  more than 50 years9
  * recent progress10,11,12,13,14,
  * hE existing methods fall far short of atomic accuracy, especially when no
    homologous structure is available
* we provide the first computational method that can regularly predict protein
  structures with atomic accuracy even in cases in which no similar structure
  is known
* AlphaFold validated in the challenging 14th Critical Assessment of protein
  Structure Prediction (CASP14)15, demonstrating
  accuracy competitive with experimental structures in a majority of cases and
  greatly outperforming other methods.  Underpinning the latest version of
  * with a novel machine learning approach that
    incorporates physical and biological knowledge about protein structure,
    leveraging multi-sequence alignments, into the design of the deep lear algo
